game/missiongenerator/aircraft/waypoints/recoverytanker.py

```diff
@@ -0,0 +1,88 @@
+from dcs.point import MovingPoint
+from dcs.task import (
+    ActivateBeaconCommand,
+    RecoveryTanker,
+    Tanker,
+    SetUnlimitedFuelCommand,
+)
+
+from game.ato import FlightType
+from game.utils import feet, knots
+from .pydcswaypointbuilder import PydcsWaypointBuilder
+
+
+class RecoveryTankerBuilder(PydcsWaypointBuilder):
+    def add_tasks(self, waypoint: MovingPoint) -> None:
+
+        assert self.flight.flight_type == FlightType.REFUELING
+
+        # Unlimited fuel option : disable at racetrack start. Must be first option to work.
+        if self.flight.squadron.coalition.game.settings.ai_unlimited_fuel:
+            if waypoint.tasks and isinstance(
+                waypoint.tasks[0], SetUnlimitedFuelCommand
+            ):
+                waypoint.tasks[0] = SetUnlimitedFuelCommand(False)
+            else:
+                waypoint.tasks.insert(0, SetUnlimitedFuelCommand(False))
+
+        clouds = self.flight.squadron.coalition.game.conditions.weather.clouds
+        waypoint.add_task(Tanker())
+        group_id = self._get_carrier_group_id()
+        speed = knots(250).meters_per_second
+        altitude = feet(6000).meters
```

Comment:
Keep in mind that this happens during mission generation time, so this means you won't see the actual altitude which will be used during generation in the waypoints tab. Ideally we initialize this during creation of the FlightWaypoint, although it's not _that_ big of a deal 😅

Reply:
I see where to add it to the waypoints tab but then does the altitude set here get overwritten or?

Reply:
The other way around, i.e. whatever you would set in the waypoints tab would get overwritten by this part. Does that make sense? 😅